dune_client/api/query.py

```diff
@@ -0,0 +1,137 @@
+
+    def create_query(
+        self,
+        name: str,
+        query_sql: str,
+        params: Optional[list[QueryParameter]] = None,
+        is_private: bool = False,
+    ) -> DuneQuery:
+        """
+        Creates Dune Query by ID
+        https://dune.com/docs/api/api-reference/edit-queries/create-query/
+        """
+        parameters = {
+            "name": name,
+            "query_sql": query_sql,
+            "private": is_private,
+        }
+        if params is not None:
+            parameters["parameters"] = [p.to_dict() for p in params]
+        response_json = self._post(route="/query/", params=parameters)
+        try:
+            query_id = int(response_json["query_id"])
+            # Note that this requires an extra request.
+            return self.get_query(query_id)
+        except KeyError as err:
+            raise DuneError(response_json, "create_query Response", err) from err
+
+    def get_query(self, query_id: int) -> DuneQuery:
+        """
+        Retrieves Dune Query by ID
+        https://dune.com/docs/api/api-reference/edit-queries/get-query/
+        """
+        response_json = self._get(route=f"/query/{query_id}")
+        return DuneQuery.from_dict(response_json)
+
+    def update_query(  # pylint: disable=too-many-arguments
+        self,
+        query_id: int,
+        name: Optional[str] = None,
+        query_sql: Optional[str] = None,
+        params: Optional[list[QueryParameter]] = None,
+        description: Optional[str] = None,
+        tags: Optional[list[str]] = None,
+    ) -> int:
+        """
+        Updates Dune Query by ID
+        https://dune.com/docs/api/api-reference/edit-queries/update-query
+
+        The request body should contain all fields that need to be updated.
+        Any omitted fields will be left untouched.
+        If the tags or parameters are provided as an empty array,
+        they will be deleted from the query.
+        """
+        parameters: dict[str, Any] = {}
+        if name is not None:
+            parameters["name"] = name
+        if description is not None:
+            parameters["description"] = description
+        if tags is not None:
+            parameters["tags"] = tags
+        if query_sql is not None:
+            parameters["query_sql"] = query_sql
+        if params is not None:
+            parameters["parameters"] = [p.to_dict() for p in params]
+
+        if not bool(parameters):
+            # Nothing to change no need to make reqeust
+            self.logger.warning("called update_query with no proposed changes.")
+            return query_id
+
+        response_json = self._patch(
+            route=f"/query/{query_id}",
+            params=parameters,
+        )
+        try:
+            # No need to make a dataclass for this since it's just a boolean.
+            return int(response_json["query_id"])
+        except KeyError as err:
+            raise DuneError(response_json, "update_query Response", err) from err
```

Comment:
Same here too

Reply:
In this case, there is literally a comment saying why we didn't bother making a response class... 

Anyway, if we think its overall cleaner we could go with that. I believe its mostly that these routes were added over a year after the others an the general flow had been forgotten.